pkg/clients/gcp.go

```diff
@@ -106,12 +108,27 @@ func UseProviderConfig(ctx context.Context, c client.Client, mg resource.Managed
 		if err != nil {
 			return "", nil, errors.Wrap(err, "cannot get credentials")
 		}
-		opts = append(opts, option.WithCredentialsJSON(data))
+		if isJSON(data) {
+			opts = append(opts, option.WithCredentialsJSON(data))
+		} else {
+			t := oauth2.Token{
+				AccessToken: string(data),
+			}
+			if ok := t.Valid(); !ok {
+				return pc.Spec.ProjectID, opts, errors.New("Access token invalid")
+			}
+			ts := oauth2.StaticTokenSource(&t)
+			opts = append(opts, option.WithTokenSource(ts))
+		}
```

Comment:
I think you accidentally reverted the removal of the `else` block in the Rebase commit. Can you remove it again?

Reply:
Removed the else and return after the JSON check.